### PR TITLE
[pytorch][test][sagemaker] Allow py38 option on PT SM tests

### PR DESCRIPTION
**Pull request by saimidu**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Allow `py38` as input python version on SM Tests for PT Training and PT Inference

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by SergTogul — APPROVED** [review]

**Review by tejaschumbalkar — APPROVED** [review]

File: test/sagemaker_tests/pytorch/inference/conftest.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def pytest_addoption(parser):
     parser.addoption('--docker-base-name', default='pytorch')
     parser.addoption('--region', default='us-west-2')
     parser.addoption('--framework-version', default='')
-    parser.addoption('--py-version', choices=['2', '3'], default=str(sys.version_info.major))
+    parser.addoption('--py-version', choices=['2', '3', '37', '38'], default=str(sys.version_info.major))
```
Comment (arjkesh) [inline test/sagemaker_tests/pytorch/inference/conftest.py:108]: Is it possible for us to remove the dependency on minor python versions going forward? It's tough to scale if we're finding out about these types of issues late

Comment (arjkesh) [reply test/sagemaker_tests/pytorch/inference/conftest.py:108]: ^ not merge blocker

Comment (saimidu) [reply test/sagemaker_tests/pytorch/inference/conftest.py:108]: Agree. Let's make a PR, later, to remove this unnecessary constraint.
